### DISCUSSION TITLE
any means to expose shader errors?

### DISCUSSION THREAD

**Discussion by tiye** [greggman/twgl.js#209] [Q&A] [answered]
I want to expose shader compilation errors as string and display them in my own UI. I looked into twgl and it does not seem to expose those messages.
https://github.com/greggman/twgl.js/blob/95e5c565a1baee519a60d679fbec1e99ec46ccbe/src/programs.js#L582-L598

The errors are logged in the console, it's still readable for me. But is there any means that we can grab the messages in my logics?

Maybe I copy and use my own version of `twgl.createProgramInfo`, but it seems like a headache in maintanence.

**Comment by greggman** [comment, marked as answer]
```
twgl.createProgramInfo(gl, [vs, fs], { errorCallback: (msg) => { alert(msg); }});
```

**Reply by tiye** [reply]
thanks. didn't notice it's already there.

<img width="993" alt="image" src="https://user-images.githubusercontent.com/449224/192563907-fd693585-fff0-4840-abbe-5c9cf9ecdee1.png">
